### PR TITLE
Constify parallelinformation and workaround Eigen's weirdness

### PR DESCRIPTION
**Pull request by blattms**
* We need to compute the reductions on a const object and therefore this PR makes them const
* Eigen's containers have nor begin()/end() iterator access for whatever reason therefore we need
to rely on operator[] *sigh*

### CODE REVIEW

File: opm/core/linalg/ParallelIstlInformation.hpp [hunk 1]
```diff
@@ -263,30 +263,31 @@ class ParallelISTLInformation
...
                           std::tuple<BinaryOperators...>& operators,
-                          std::tuple<ReturnValues...>& values)
+                          std::tuple<ReturnValues...>& values) const
     {
         const auto& container = std::get<I>(containers);
         if( container.size() )
         {
             auto& reduceOperator  = std::get<I>(operators);
-            auto newVal = container.begin();
+            // F*#$&ing Eigen:Block does not support STL iterators!!!!
```
Comment (atgeirr) [inline opm/core/linalg/ParallelIstlInformation.hpp]: Please do not use inappropriate language in the source code.

**Comment by atgeirr** [comment]
Other than the minor issue noted, this looks fine to me. I assume you have tested this on your system with MPI? I agree that it is strange that Eigen does not have begin() and end() members for 1d data structures, but it probably stems from being matrix-centric: should the iterator you get iterate over all elements, rows or columns? We could make a workaround though: create global begin() and end() functions taking the 1d arrays as arguments and returning pointers as iterators. This will also allow use of range-for in the future. I am not sure though if global begin() and end() are defined on gcc 4.4.

File: opm/core/linalg/ParallelIstlInformation.hpp (hunk 1 above)
Comment (blattms) [inline opm/core/linalg/ParallelIstlInformation.hpp]: Rebased with improved impartial comment.

Comment (atgeirr) [inline opm/core/linalg/ParallelIstlInformation.hpp]: Strangely, it still looks identical to me.

**Pull request merged**